controllers/store.js

```diff
@@ -1,34 +1,94 @@
-var getAllBooks = (req, res) => {
+if (process.env.NODE_ENV !== "production") {
+    require('dotenv').config()
+}
+const db_url = process.env.DB_URI
+
+const mongoose = require('mongoose');
+const Book = require("../models/book");
+const User = require("../models/user");
+const Bookcopy = require("../models/bookCopy");
+
+mongoose.connect(db_url, { useNewUrlParser: true, useUnifiedTopology: true })
+    .then(() => {
+        console.log("MONGO CONNECTION OPEN!!!")
+    })
+    .catch(err => {
+        console.log("OH NO MONGO CONNECTION ERROR!!!!")
+        console.log(err)
+    })
+
+
+var getAllBooks = async function (req, res) {
     //TODO: access all books from the book model and render book list page
-    res.render("book_list", { books: [], title: "Books | Library" });
+    const books = await Book.find({});
+    res.render("book_list", { books, title: "Books | Library" });
 }
 
-var getBook = (req, res) => {
+var getBook = async (req, res) => {
     //TODO: access the book with a given id and render book detail page
+    const { id } = req.params;
+    const book = await Book.findById(id);
+    res.render("book_detail", { book, title: "Book Details" })
 }
 
-var getLoanedBooks = (req, res) => {
+var getLoanedBooks = async (req, res) => {
 
     //TODO: access the books loaned for this user and render loaned books page
+    // const {user}=req;
+
+    const { user } = req;
+    const copies = await User.findById(user._id).populate({ path: "loaned_books", populate: { path: "book" } });
+    const books = copies.loaned_books;
+    res.render("loaned_books", { books: books, title: "Loaned Books" });
 }
 
-var issueBook = (req, res) => {
-    
+var issueBook = async (req, res) => {
+
     // TODO: Extract necessary book details from request
     // return with appropriate status
     // Optionally redirect to page or display on same
+    const { user, body } = req;
+    const books = await Bookcopy.find({ book: body.bid });
+    let id;
+    for (let bookcopy of books) {
+        if (bookcopy.status === true) {
+            id = bookcopy._id;
+            await Book.findByIdAndUpdate(body.bid, { $inc: { available_copies: -1 } });
+            await Bookcopy.findByIdAndUpdate(id, { status: false, borrow_date: Date.now(), borrower: user.id });
```

Comment:
Modify `borrow_date` to only show 'Mon Aug 23 2021 16:56:09 GMT' (aka make date more readable for user)